web/src/js/components/Modes/Regular.tsx

```diff
@@ -11,6 +11,12 @@ export default function Regular() {
         (state) => state.modes.regular,
     );
 
+    const error_state = useAppSelector((state) => {
+        if (state.backendState.servers[0]) {
```

Comment:
Right now servers is an array with an undefined order, we can't expect `[0]` to be regular mode. :)

Reply:
Ahhh right! Could we change that? something like `state.backendState.servers[regular]...` ?

Reply:
Which solution is better in your opinion?
```
export interface ServerInfo {
    description: string;
    full_spec: string;
    is_running: boolean;
    last_exception: string | null;
    listen_addrs: [string, number][];
    //type: string;
    wireguard_conf?: string;
}

export interface BackendState {
    available: boolean;
    version: string;
    contentViews: string[];
    servers: { [type: string] : ServerInfo};
}
```
or
```
export interface ServerInfo {
    description: string;
    full_spec: string;
    is_running: boolean;
    last_exception: string | null;
    listen_addrs: [string, number][];
    type: string;
    wireguard_conf?: string;
}

export interface BackendState {
    available: boolean;
    version: string;
    contentViews: string[];
    servers: {
        regular: ServerInfo | undefined;
        local: ServerInfo | undefined;
        wireguard: ServerInfo | undefined;
        reverse: ServerInfo | undefined;
    };
}
```
or we can keep the list and check the type 😅

Reply:
I think 1) is best. It probably should still have a `type` field because the spec can be something like `regular@8085` and we don't want to parse it to figure out it's regular mode. :)

2) falls short for multiple reverse proxies for instance.

Reply:
alright thanks! Let's go for 1! :)

Reply:
I realize now that I need to change some things on the backend as well. For this reason, does it make sense to make these changes in this PR and include them in our MVP? Is it worth it?

Reply:
we could do something like this now for each mode: 
```
const backend_error = useAppSelector((state) => {
        if (state.backendState.servers) {
            for (const server of state.backendState.servers) {
                if (server.type === "regular") {
                    console.log(server.last_exception);
                    return server.last_exception;
                }
            }
        }
        return "";
    });
```

Reply:
Let's get this one here flying with the current crappy solution, and once we're in main again we can follow up with a quick PR that changes it to 1).